openmc/deplete/results.py

```diff
@@ -70,7 +71,7 @@ def __init__(self, filename=None):
 
 
     @classmethod
-    def from_hdf5(cls, filename):
+    def from_hdf5(cls, filename: str):
```

Comment:
This could also be path-like. We should define a standard type-hint for paths that accounts for both strings and path-like, something like this:
https://github.com/watts-dev/watts/blob/9004f2d948be74352d05cb9d10573915e9f39102/src/watts/fileutils.py#L17-L18

That should probably live in another file so that it can be imported wherever needed.

Reply:
I've had a go at this in commit the latest commit